libs/vscode/project-graph/src/lib/legacy-implementation/graph.machine.ts

```diff
@@ -27,168 +34,165 @@ interface Context {
+              target: 'error',
+            },
+          ],
         },
-        viewDestroyed: {
-          entry: ['log', 'refreshData', 'clearProject'],
+      },
+      error: {
+        entry: 'log',
+      },
+      no_project: {},
+      viewReady: {
+        entry: 'log',
+        on: {
+          VIEW_DESTROYED: {
+            target: 'viewDestroyed',
+          },
         },
       },
+      viewDestroyed: {
+        entry: ['log', 'refreshData', 'clearProject'],
+      },
     },
-    {
-      services: {
-        generateContent: async (context) => {
+  }).provide({
+    actors: {
+      generateContent: fromPromise(
+        async ({ input }: { input: { messageType: string } }) => {
           const showAffected =
-            context?.project?.type === MessageType.affectedProjects;
+            input.messageType === MessageType.affectedProjects;
           return await createProjectGraph(showAffected);
-        },
-      },
-      actions: {
-        log: (context, event) => {
-          getOutputChannel().appendLine(
-            `Graph Machine - ${JSON.stringify(event)}, ${JSON.stringify(
-              context
-            )}`
-          );
-        },
-        refreshData: assign((context, event) => {
-          return {
-            state: State.init,
-          };
-        }),
-        contentLoaded: assign((context, event) => {
-          return {
-            state: State.loaded,
-            error: null,
-          };
-        }),
-        loadingFailed: assign((context, { data }) => {
-          return {
-            state: State.error,
-            error: data,
-          };
-        }),
-        loadingStarted: assign((context, event) => {
-          return {
-            state: State.loading,
-          };
-        }),
-        clearProject: assign((context, event) => {
-          return {
-            project: null,
-          };
-        }),
-        projectSelected: assign((context, { data }) => {
-          return {
-            project: {
-              projectName: data.projectName,
-              taskName: data.taskName,
-              type: data.type,
-            },
-          };
-        }),
+        }
+      ),
+    },
+    actions: {
+      log: (context, event) => {
+        getOutputChannel().appendLine(
+          `Graph Machine - ${JSON.stringify(event)}, ${JSON.stringify(context)}`
+        );
       },
-      guards: {
-        loadGraph: (context) => {
-          getOutputChannel().appendLine(`Graph guard - ${context.state}`);
-          return context.state !== State.loaded;
-        },
+      refreshData: assign((context, event) => {
+        return {
+          state: State.init,
+        };
+      }),
+      contentLoaded: assign((context, event) => {
+        return {
+          state: State.loaded,
+          error: null,
+        };
+      }),
+      loadingFailed: assign(({ context, event }) => {
+        if (event.type !== 'error.platform.loadingContent') {
```

Comment:
Why do we need this guard clause now?

Reply:
I honestly have no idea. It seems like we could remove it but there could be something I'm missing. My goal was to not mess with the legacy implementation code so I'm not sure I want to change things here.